ui/src/app/workflows/components/workflow-details/workflow-details.tsx

```diff
@@ -177,159 +177,56 @@ export class WorkflowDetails extends React.Component<RouteComponentProps<any>, W
         );
     }
 
+    private confirmAction(title: string): void {
+        if (!confirm(`Are you sure you want to ${title} this workflow?`)) {
+            return;
+        }
+        return;
+    }
+
+    private getHandleErrorFunction(title: string): () => void {
+        return () => {
+            this.appContext.apis.notifications.show({
+                content: `Unable to ${title} workflow`,
+                type: NotificationType.Error
+            });
+        };
+    }
+
+    private performAction(action: (name: string, namespace: string) => Promise<any>, title: string, redirect: string, ctx: ContextApis): void {
+        this.confirmAction(title);
+        action(this.props.match.params.name, this.props.match.params.namespace)
+            .then(() => ctx.navigation.goto(uiUrl(redirect)))
+            .catch(this.getHandleErrorFunction(title));
+    }
+
     private getItems(workflowPhase: NodePhase, ctx: any) {
-        const items = [
-            {
-                title: 'Retry',
-                iconClassName: 'fa fa-undo',
-                disabled: workflowPhase === undefined || !(workflowPhase === 'Failed' || workflowPhase === 'Error'),
-                action: () => this.retryWorkflow(ctx)
-            },
-            {
-                title: 'Resubmit',
-                iconClassName: 'fa fa-plus-circle ',
-                action: () => this.resubmitWorkflow(ctx)
-            },
-            {
-                title: 'Suspend',
-                iconClassName: 'fa fa-pause',
-                disabled: !Utils.isWorkflowRunning(this.state.workflow) || Utils.isWorkflowSuspended(this.state.workflow),
-                action: () => this.suspendWorkflow(ctx)
-            },
-            {
-                title: 'Resume',
-                iconClassName: 'fa fa-play',
-                disabled: !Utils.isWorkflowSuspended(this.state.workflow),
-                action: () => this.resumeWorkflow(ctx)
-            },
-            {
-                title: 'Stop',
-                iconClassName: 'fa fa-stop-circle',
-                disabled: !Utils.isWorkflowRunning(this.state.workflow),
-                action: () => this.stopWorkflow(ctx)
-            },
-            {
-                title: 'Terminate',
-                iconClassName: 'fa fa-times-circle',
-                disabled: !Utils.isWorkflowRunning(this.state.workflow),
-                action: () => this.terminateWorkflow(ctx)
-            },
-            {
-                title: 'Delete',
-                iconClassName: 'fa fa-trash',
-                action: () => this.deleteWorkflow(ctx)
-            }
-        ];
+        const actions: any = Actions.WorkflowActions;
+        const items = Object.keys(actions).map(actionName => {
```

Comment:
Are we sure that this returns the items in the same order every time? We don't want to have the buttons shuffle around.

Reply:
I'm fairly confident that as of ES2015 the order of object props can be guaranteed, and as of ES6 `Object.keys` relies on the underlying method that they use to guarantee the order. Here's the relevant portion of the spec: 
https://tc39.es/ecma262/#sec-ordinaryownpropertykeys

Reply:
Great! Thanks